### ISSUE TITLE
Update to Java 8 !

### ISSUE THREAD

**Issue by rautamiekka** [WeTheInternet/collide#4]
Simple as that. There are absolutely no reasons to stick to Java 7, let alone code in a way which is JRE version-dependent.

**Comment by JamesXNelson** [comment]
Hm.  My latest branch has java 8 support (and I've already done java 9 support for gwt itself).

All the java 7 bits are leftover from the project inherited from google. 

I have ripped out all the ant, upgraded to gradle, removed binary jars from source checkin, and now I just need to release my gwt fork / dependencies, and you will be able to build again with java 8!  It would probably build with java 9 if you are willing to do some unholy things; proper support for that won't be coming for a few more months.

**Comment by mkatychev** [comment]
@JamesXNelson would it be possible to do a quick writeup on how to build with gradle for those of us not familiar with java? running `gradle build` gets me:
```
> Could not resolve all artifacts for configuration ':api:classpath'.
   > Could not find fr.putnami.gwt:putnami-gradle-plugin:0.5.0-SNAPSHOT.
     Searched in the following locations:
       - file:/Users/$USER/.m2/repository/fr/putnami/gwt/putnami-gradle-plugin/0.5.0-SNAPSHOT/maven-metadata.xml
       - file:/Users/$USER/.m2/repository/fr/putnami/gwt/putnami-gradle-plugin/0.5.0-SNAPSHOT/putnami-gradle-plugin-0.5.0-SNAPSHOT.pom
       - file:/Users/$USER/.m2/repository/fr/putnami/gwt/putnami-gradle-plugin/0.5.0-SNAPSHOT/putnami-gradle-plugin-0.5.0-SNAPSHOT.jar
       - https://jcenter.bintray.com/fr/putnami/gwt/putnami-gradle-plugin/0.5.0-SNAPSHOT/maven-metadata.xml
       - https://jcenter.bintray.com/fr/putnami/gwt/putnami-gradle-plugin/0.5.0-SNAPSHOT/putnami-gradle-plugin-0.5.0-SNAPSHOT.pom
       - https://jcenter.bintray.com/fr/putnami/gwt/putnami-gradle-plugin/0.5.0-SNAPSHOT/putnami-gradle-plugin-0.5.0-SNAPSHOT.jar
     Required by:
         project :api


```

**Comment by JamesXNelson** [comment]
ahhh.  Sorry, I've been caught up in work and kids (I have two little ones atm).

I will take a crack at fixing up the gradle build for this over the weekend; there are some dependencies that I use locally which aren't in public repos, so I'll come up with something that manages this for you.

This whole project has been back-burnered a little while I work on the j2cl / gwt3 release; my plan is to circle back around and update to elemental 2 + xapi once j2cl / gwt3 are up and running... but time is sadly my enemy right now.  I'll leave a note on my calendar to spend an hour this weekend so others can play / take over / enjoy.

**Comment by JamesXNelson** [comment]
Also, @mkatychev if you are not familiar w/ java, then this is definitely not the project for you (java transpiled to javascript).  Internally collIDE just uses codemirror for the editor, so if that's what you're after, you'd be much better off going to the source.  I've also successfully used both ace and monaco in my day job (I usually wind up working on custom languages / polyglot runtimes).  

If it's the realtime-multiuser feature you're after, then it might be worth it for you to dig in and try, but there is no community support here (yet).  This project had a team at google, who were let go when they closed the Atlanta office, then I picked it up as a pet project, but it sadly lands pretty far to the back of my list of things to spend time on :-/

**Comment by mkatychev** [comment]
@JamesXNelson  yeah the multiuser feature is what I was after primarily, thanks for the update.
